### PR TITLE
Backport PloneFormGen migration to 1.x

### PR DESCRIPTION
**Pull request by thomasmassmann**
Uses the existing migration and adjusts some imports and method calls to be compatible with 1.x.

It also sets the correct description for fieldsets (needs to be an attribute, not a property).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,6 @@ install:
   - bin/buildout -t 5 -Nq
 before_script:
   - "export DISPLAY=:99.0"
-  - "sh -e /etc/init.d/xvfb start"
+  # - "sh -e /etc/init.d/xvfb start"
```
Comment (thomasmassmann) [inline .travis.yml:14]: Deactivated since only unit tests are run and travis-ci shuts down in 4 days.

File: collective/easyform/migration/pfg.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        self.log = StringIO()
+        handler = logging.StreamHandler(self.log)
+        logger.addHandler(handler)
+        formatter = logging.Formatter(
+            '%(asctime)s %(levelname)s %(name)s %(message)s')
+        handler.setFormatter(formatter)
+
+        alsoProvides(self.request, IDisableCSRFProtection)
+        portal = getSite()
+        migrate(portal, PloneFormGenMigrator)
```
Comment (mauritsvanrees) [inline collective/easyform/migration/pfg.py:97]: I got a `LinkIntegrityNotificationException` here. So the code should switch off linkintegrity checks first.
This works for me:

```
        portal = getSite()

        # Switch linkintegrity off temporarily.
        ptool = getToolByName(self.context, "portal_properties")
        site_props = getattr(ptool, 'site_properties', None)
        link_integrity = False
        if site_props and site_props.hasProperty(
                'enable_link_integrity_checks'):
            link_integrity = site_props.getProperty(
                'enable_link_integrity_checks', False)
            if link_integrity:
                site_props.manage_changeProperties(
                    enable_link_integrity_checks=False)

        migrate(portal, PloneFormGenMigrator)

        # switch linkintegrity back on, if needed
        if link_integrity:
            site_props.manage_changeProperties(
                enable_link_integrity_checks=True
            )
```

**Review by mauritsvanrees — CHANGES_REQUESTED** [review]
I have two inline comments that should be addressed, about compatibility when `plone.app.contenttypes` is not available, and about linkintegrity.

Other than that, looks good. I tried it with fixes for those two points, and it worked on a site with three PloneFormGen forms.

Well, for one of them I got a lot of warnings: "Number of columns does not match. Skipping row 0 in data adapter". But that is probably already a problem in the old form, because it worked fine for one of the other two forms.

And for another, I got an error when filling in the resulting easyform: `NameError: global name 'ploneformgen' is not defined`. This is in a custom script adapter which excpects that `ploneformgen` is available as variable pointing to the form, but it should now be `easyform`. It would be nice if the migration would catch this. But this PR is a back port from master, and this one would likely fail on master as well. So: fine to ignore this detail.

Summary: please address my two inline comments. After that, it can be merged and I could do a release.

**Comment by mauritsvanrees** [comment]
FWIW, I quickly created a new PFG form with a fieldset. This looked ugly in PFG itself, but never mind that. After migration, it showed up as a proper fieldset in a new tab. So this new part of the migration seems fine. I cannot test how it would have gone without that part of the fix though.

**Comment by mauritsvanrees** [comment]
@thomasmassmann Is this still on your radar?

**Comment by thomasmassmann** [comment]
> @thomasmassmann Is this still on your radar?

@mauritsvanrees yes, i will add your changes and test it again with my initial setup.

**Pull request merged**